src/Server.cpp

```diff
@@ -358,6 +358,15 @@ void server_handler(int _fd_min, int _fd_max, int _fd_num) {
 void server_handler(handler_info *p_info)
 {
 	if (p_info) {
+#ifdef USING_VMA_EXTRA_API
+		if (g_vma_api) {
+			g_zero_data.pkt_buf = (unsigned char*)MALLOC(Message::getMaxSize());
+			if (g_zero_data.pkt_buf == NULL) {
+				log_err("Failed to allocate g_pkt_buf");
+				return;
+			}
+		}
+#endif
```

Comment:
this low level code makes the _server_handler_ function to be ugly since without it it looks clean - dealing only with high level logic that can be understood at a glance.
You should create new general purpose initialization function and place your code inside it